### PR TITLE
fixed the path file recursive query issue

### PR DESCRIPTION
**Pull request by blackWins**

### CODE REVIEW

**Comment by blackWins** [comment]
#223

File: src/AbpHelper.Core/Steps/StepWithOption.cs [hunk 1]
```diff
@@ -109,12 +109,12 @@ public WorkflowExpression<bool> CommandOption
                     matchedSln = true;
                 }
 
-                var result = searchFunc(directory, pattern, SearchOption.TopDirectoryOnly).FirstOrDefault();
-                if (!result.IsNullOrWhiteSpace())
+                var result = searchFunc(directory, pattern, SearchOption.TopDirectoryOnly);
+                if (result.Any())
```
Comment (gdlcf88) [inline src/AbpHelper.Core/Steps/StepWithOption.cs:113]: `if (result is not null)`

Comment (blackWins) [reply src/AbpHelper.Core/Steps/StepWithOption.cs:113]: image

the `Directory.EnumerateFiles`  return `IEnumerable<string>` .`result is not null` will alway return `ture`

Comment (blackWins) [reply src/AbpHelper.Core/Steps/StepWithOption.cs:113]: image

Comment (gdlcf88) [reply src/AbpHelper.Core/Steps/StepWithOption.cs:113]: Yep, `Directory.EnumerateFiles` will never return null, but `searchFunc` will. So please check the null value.

Comment (gdlcf88) [reply src/AbpHelper.Core/Steps/StepWithOption.cs:113]: `result is not null` will cause a warning in the IDE if the nullable is enabled for this project.

Now I think this "if" line can just be removed.

Comment (blackWins) [reply src/AbpHelper.Core/Steps/StepWithOption.cs:113]: the `FindInDirectoryInQueue()` maybe have same probleam.
`line:149 var result = searchFunc(directory, pattern, SearchOption.TopDirectoryOnly).FirstOrDefault();`

Comment (blackWins) [reply src/AbpHelper.Core/Steps/StepWithOption.cs:113]:
```
private static string? FindInDirectoryInQueue(Queue<string> queue, string pattern,
            IReadOnlySet<string> actualExcluded, Func<string, string, SearchOption, IEnumerable<string>> searchFunc,
            bool includeSubModules = false)
        {
            var matchedSln = false;
            while (queue.TryDequeue(out var directory))
            {
                if (!includeSubModules &&
                    null != searchFunc(directory, "*.sln", SearchOption.TopDirectoryOnly).FirstOrDefault())
                {
                    if (matchedSln)
                    {
                        continue;
                    }

                    matchedSln = true;
                }

                //var result = searchFunc(directory, pattern, SearchOption.TopDirectoryOnly).FirstOrDefault();
                //if (!result.IsNullOrWhiteSpace())
                //{
                //    return result;
                //}
                
                var result = searchFunc(directory, pattern, SearchOption.TopDirectoryOnly);
                if (result != null)
                {
                    return result.FirstOrDefault();
                }

                foreach (var d in Directory.EnumerateDirectories(directory))
                {
                    if (actualExcluded.Contains(d))
                    {
                        continue;
                    }

                    queue.Enqueue(d);
                }
            }

            return null;
        }
```

**Pull request merged**